content/en/blog/2021/simple-vms-part-2/index.md

```diff
@@ -0,0 +1,290 @@
+---
+title: "Part 2 - Implementing an easier way to add virtual machines to Istio service mesh"
+description: How to statically configure envoy to join a mesh.
+publishdate: 2022-01-17
+attribution: "Nick Nellis (Solo.io)"
+keywords: [virtualmachine,Istio,networking,envoy]
+---
+![Virtual Machine](./istio-setup.png)
+
+Part 1 - [An easier way to add virtual machines to Istio service mesh](/blog/2021/simple-vms/)
+
+A lot of Istio users would like to connect their virtual machines to their service mesh but the current process is very cumbersome and hard to automate. This series explains a simpler way to connect outside applications securely to your applications running in a service mesh. If you would like to know more about why this might work for you, check out part 1 here.
+
+## Preface
+
+As explained in [part 1](/blog/2021/simple-vms/), to reduce complexity of the virtual machine initialization we will not be registering it with Istio. Instead we will be issuing a certificate directly to the virtual machine and manually configuring the envoy instance to communicate inside the mesh via the east/west gateway. First we will need to organize our certificates so that the service mesh applications will trust these external workloads.
+
+## Certificate Management
+
+Since Istio will not be issuing the certificates for the virtual machines, it will need to be created and managed by some other means. In order for Istio to trust the traffic coming from the virtual machines, the certificates issued must be rooted in the same chain of trust as the service mesh applications.
+
+![Certificate Management](./cert-architecture.png)
+
+The simplest way to organize your certificates is to create a shared root CA. Then two intermediate CA certificates can be issued and used for creating Istio workload certificates and one for virtual machine certificates. The virtual machine CA could easily be managed by a third party certificate management system like Hashicorp's Vault.
+
+## Istio Setup and Deployment
+
+![Istio Setup Deployment](./istio-setup.png)
+
+The setup for Istio to accept outside secured connections is very trivial. Using the IstioOperator configuration an east/west gateway needs to be created and configured to pass through connections.
+
+### Creating the east/west gateway
+
+Below is an example of an Istio installation of just istiod and the east/west gateway shown above. `PILOT_SKIP_VALIDATE_TRUST_DOMAIN` is enabled so that the virtual machines can have their own SPIFFE ID format.
```

Comment:
```suggestion
Below is an example of an Istio installation of just istiod and the east-west gateway shown above. `PILOT_SKIP_VALIDATE_TRUST_DOMAIN` is enabled so that the virtual machines can have their own SPIFFE ID format.
```